XMonad/Hooks/DynamicStatusBarConfigs.hs

```diff
@@ -0,0 +1,125 @@
+{-# LANGUAGE TupleSections #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  XMonad.Hooks.DynamicStatusBarConfigs
+-- Copyright   :  (c) Yecine Megdiche <yecine.megdiche@gmail.com>
+-- License     :  BSD3-style (see LICENSE)
+--
+-- Maintainer  :  Yecine Megdiche <yecine.megdiche@gmail.com>
+-- Stability   :  unstable
+-- Portability :  unportable
+--
+-- This module provides dynamic status bars, that react to screen changes.
+-- Heavily inspired by "XMonad.Hooks.DynamicBars".
+----------------------------------------------------------------------------
+module XMonad.Hooks.DynamicStatusBarConfigs
+  (
+    -- * Usage
+    -- $usage
+    dynamicSBs
+  )
+where
+
+
+import           Data.List                      ( partition
+                                                , (\\)
+                                                )
+import           Data.Foldable                  ( traverse_ )
+import           Data.Monoid                    ( All(..) )
+import           Graphics.X11.Xrandr            ( xrrSelectInput )
+
+import           XMonad
+import           XMonad.Hooks.DynamicLog
+import qualified XMonad.StackSet               as W
+import qualified XMonad.Util.ExtensibleState   as XS
+
+-- $usage
+-- Using multiple status bars just with "XMonad.Hooks.DynamicLog' works well
+-- as long as the screen configuration does not change often. Using
+-- 'dynamicSBs' fixes that problem: by providing a function that creates
+-- status bars, it takes care of setting up the event hook, the log hook
+-- and the startup hook necessary to make the status bars, well, dynamic.
+--
+-- > mainTopBar, mainBottomBar, secondaryBar1 :: IO StatusBarConfig
+-- > mainTopBar = statusBarPropConfig "xmobar -x 0 $HOME/.config/xmobar/xmobarrc_top" ppTop
+-- > mainBottomBar = statusBarHandleConfig "xmobar -x 0 $HOME/.config/xmobar/xmobarrc_bottom" ppBottom
+-- > secondaryBar1 = statusBarPropConfig "xmobar -x 1 $HOME/.config/xmobar/xmobarrc1" pp1
+-- >
+-- > barSpawner :: ScreenId -> IO StatusBarConfig
+-- > barSpawner 0 = mainTopBar <> mainBottomBar -- two bars on the main screen
+-- > barSpawner 1 = secondaryBar1
+-- > barSpawner _ = memtpy -- nothing on the rest of the screens
+-- >
+-- > main = xmonad $ dynamicSBs barSpawner (def { ... })
+
+newtype ActiveSBs = ASB {getASBs :: [(ScreenId,  StatusBarConfig)]}
+
+instance ExtensionClass ActiveSBs where
+  initialValue = ASB []
+
+-- | Given a function to create status bars, 'dynamicSBs'
+-- adds the dynamic status bar capabilities to the config.
+dynamicSBs :: (ScreenId -> IO StatusBarConfig) -> XConfig l -> XConfig l
+dynamicSBs f conf = conf
+  { startupHook     = startupHook conf
+                      *> setupEventHandler
+                      *> cleanAll
+                      *> updateSBs f
+  , logHook         = logHook conf *> logSBs
```

Comment:
Just curious: why `*>` in here and `<>` in `handleEventHook`?

Reply:
I thought of the `logHook` and `startupHook` more like a sequence of actions, and `handleEventHook` as a single action that is composed.. I'll make them the same

Reply:
Oh, that makes sense. Although I must admit that `*>` invokes more of a "give me the result on the right" vibe rather than "sequencing", for which `>>` is more common, at least among those who learned Haskell before `*>` was a thing. :-)

Reply:
I guess, I was influenced by the choices @slotThe  made in the original `makeStatusBar` and it kinda stuck with me.. I prefer `<>` though, but `>>` is more common

Reply:
> at least among those who learned Haskell before `*>` was a thing. :-)

Coming from someone who did not, I pretty much treat `>>` as a deprecated versin of `*>` (much like `liftM` -> `fmap`, `mapM` -> `traverse`, `return` -> `pure`, ... we really have a lot of those now) so both convey a sequencing vibe to me.  I do think that `<>` invokes more of a "this is a single thing" instead of "these are different, mutually independent steps" feeling